indico/testing/fixtures/editing.py

```diff
@@ -0,0 +1,84 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2023 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+import pytest
+
+from indico.modules.events.editing.models.editable import Editable, EditableType
+from indico.modules.events.editing.models.file_types import EditingFileType
+from indico.modules.events.editing.models.revision_files import EditingRevisionFile
+from indico.modules.events.editing.models.revisions import EditingRevision, FinalRevisionState, InitialRevisionState
+from indico.util.date_time import now_utc
+
+
+@pytest.fixture
+def create_editable(db):
+    """Return a callable which lets you create editables."""
+    def _create_editable(contribution, type, **kwargs):
+        editable = Editable(contribution=contribution, type=type, **kwargs)
+        db.session.flush()
+        return editable
+    return _create_editable
+
+
+@pytest.fixture
+def dummy_editable(dummy_contribution, create_editable):
+    """Create a dummy editable."""
+    return create_editable(dummy_contribution, EditableType.paper, id=42)
+
+
+@pytest.fixture
+def create_editing_revision(db):
+    """Return a callable which lets you create editing revisions."""
+    def _create_editing_revision(editable, submitter, initial_state, final_state, reviewed_dt, **kwargs):
+        revision = EditingRevision(editable=editable, submitter=submitter,
+                                   initial_state=initial_state,
+                                   final_state=final_state, reviewed_dt=reviewed_dt, **kwargs)
+        db.session.flush()
+        return revision
+    return _create_editing_revision
+
+
+@pytest.fixture
+def dummy_editing_revision(dummy_user, dummy_editable, create_editing_revision):
+    """Create a dummy editing revision."""
+    return create_editing_revision(dummy_editable, submitter=dummy_user, id=42,
+                                   initial_state=InitialRevisionState.ready_for_review,
+                                   final_state=FinalRevisionState.accepted, reviewed_dt=now_utc())
+
+
+@pytest.fixture
+def create_editing_file_type(db):
+    """Return a callable which lets you create editing file types."""
+    def _create_editing_file_type(name, extensions, type, **kwargs):
+        file_type = EditingFileType(name=name, extensions=extensions, type=type, **kwargs)
+        db.session.add(file_type)
+        db.session.flush()
+        return file_type
+    return _create_editing_file_type
+
+
+@pytest.fixture
+def dummy_editing_file_type(create_editing_file_type):
+    """Create a dummy editing file type."""
+    return create_editing_file_type(id=42, name='TXT', extensions=['txt'], type=EditableType.paper)
```

Comment:
I think all those hardcoded (but still somewhat low) IDs are a bit dangerous since if you were to create 42 objects in a test, you'd get a collision w/ the dummy one.